haskell-overlays/disable-template-haskell.nix

```diff
@@ -0,0 +1,61 @@
+{ haskellLib, fetchFromGitHub }:
+
+let
+  # TODO upstream into Haskell lib
+  dontUseCustomSetup = p: haskellLib.overrideCabal p (drv: {
+    preCompileBuildDriver = assert (drv.preCompileBuildDriver or null) == null; ''
+      rm Setup.hs || rm Setup.lhs
```

Comment:
Not all projects have either of these. Cabal works fine without them.

Reply:
Right now I'm just trying to move code around, not/minimally changing hashes etc. We can do this and other such refactors in a follow-up PR.

Reply:
👍

Reply:
In a lot of cases, we've applied big hammers where probably a smaller one would do, in the interest of getting mobile/etc. done as quickly as possible.  If you have the time/energy, helping turn them into smaller ones (and testing to make sure everything still works!) would be greatly appreciated :)

Reply:
Great advice. Thanks!